### PR TITLE
Add Upgrade transaction for Staking contract

### PR DESCRIPTION
**Pull request by joshuahannan**
Adds contract, transaction, and arguments for the upgrade to the FlowIDTableStaking contract on mainnet

### CODE REVIEW

**Review by peterargue — COMMENTED** [review]
looks like there are a few differences between the PR (https://github.com/onflow/flow-core-contracts/pull/321) and `transactions/update-contract/2023/feb-1/FlowIDTableStaking.cdc`. don't forget to pull those in

File: transactions/update-contract/2023/feb-1/upgrade_nodelists.cdc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        // `getCurrentRoleNodeCounts` calculates the counts based on the current list
+        // so we just store the result in storage after calculating
+        Let slotCounts = FlowIDTableStaking.getCurrentRoleNodeCounts()
+        stakingAccount.save(slotCounts, to: /storage/flowStakingRoleNodeCounts)
+
+        // Approve List needs to be changed from [String] to {String: Bool}
+        let approveList: [String] = stakingAccount.load<[String]>(from: /storage/idTableApproveList)
+            ?? panic("Could not load the approve list from storage")
+
+        // Create the approve list dictionary with the loaded value, then store the dictionary in the same path
```
[the remark this answers is not in the document]
Comment (joshuahannan) [reply transactions/update-contract/2023/feb-1/upgrade_nodelists.cdc:46]: yep, this was just a draft. I've tested the transaction now and updated it

**Comment by joshuahannan** [comment]
@vishalchangrani @Kay-Zee I've added the instructions to the README for doing the multisign with the staking account and service account. Can y'all take a look and let me know if the instructions look good? Thanks!

File: transactions/update-contract/2023/feb-8/README.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+4. Checkout this repo locally.
+5. Open a terminal and go into the directory `service-account`
+
+## Dapper Labs Builds
+
+```sh
+flow transactions build ./templates/upgrade_nodelists.cdc \
+  --config-path flow-staking.json \
+  --network mainnet \
+  --args-json "$(cat "./transactions/update-contract/2023/feb-8/arguments-update-contract-FlowIDTableStaking-mainnet.json")" \
```
Comment (vishalchangrani) [inline transactions/update-contract/2023/feb-8/README.md:46]: https://github.com/onflow/service-account/pull/210

**Comment by vishalchangrani** [comment]
I ran the RLP generation and then tried to sign the generated transaction and ran into this issue:

`: encoding/hex: invalid byte: U+000A`.


I executed:
1. `flow transactions build ./transactions/update-contract/2023/feb-1/upgrade_nodelists.cdc   --config-path flow-staking.json   --network mainnet   --args-json "$(cat "./transactions/update-contract/2023/feb-1/arguments-update-contract-FlowIDTableStaking-mainnet.json")"   --proposer 0x8624b52f9ddcd04a   --proposer-key-index 5   --authorizer 0x8624b52f9ddcd04a   --payer 0xe467b9dd11fa00df   --gas-limit 1000000   -x payload   --save ./transactions/update-contract/2023/feb-1/upgrade-nodelists-feb-1-unsigned.rlp`
2. `flow transactions sign ./transactions/update-contract/2023/feb-8/upgrade-nodelists-feb-8-unsigned.rlp   --config-path flow-staking.json   --signer vishal   --filter payload   --save ./transactions/update-contract/2023/feb-8/upgrade-nodelists-feb-8-sig-1.rlp`

**Comment by vishalchangrani** [comment]
> I ran the RLP generation and then tried to sign the generated transaction and ran into this issue:
> 
> `: encoding/hex: invalid byte: U+000A`.
> 
> I executed:
> 
> 1. `flow transactions build ./transactions/update-contract/2023/feb-1/upgrade_nodelists.cdc   --config-path flow-staking.json   --network mainnet   --args-json "$(cat "./transactions/update-contract/2023/feb-1/arguments-update-contract-FlowIDTableStaking-mainnet.json")"   --proposer 0x8624b52f9ddcd04a   --proposer-key-index 5   --authorizer 0x8624b52f9ddcd04a   --payer 0xe467b9dd11fa00df   --gas-limit 1000000   -x payload   --save ./transactions/update-contract/2023/feb-1/upgrade-nodelists-feb-1-unsigned.rlp`
> 2. `flow transactions sign ./transactions/update-contract/2023/feb-8/upgrade-nodelists-feb-8-unsigned.rlp   --config-path flow-staking.json   --signer vishal   --filter payload   --save ./transactions/update-contract/2023/feb-8/upgrade-nodelists-feb-8-sig-1.rlp`

@janezpodhostnik would you know?

**Review by vishalchangrani — APPROVED** [review]
I tested the command to build, sign and send the transactions. Found an issue with Flow cli which has not been fixed. Flow cli needs to be updated to v0.45.2.

**Pull request merged**